src/persist-client/src/internal/gc.rs

```diff
@@ -360,6 +368,41 @@ where
             req.shard_id, req.new_seqno_since
         );
 
+        // Finally, apply the remaining diffs and calculate the live / current
+        // set of diffs and blobs for metrics.
+        while let Some(state) = states.next() {
+            live_diffs += 1;
+            state.collections.trace.map_batches(|b| {
+                for part in b.parts.iter() {
+                    live_batch_blobs.insert(part.key.to_owned());
+                }
+            });
+        }
+
+        let state = states.state();
+
+        let live_batch_blobs: i64 = live_batch_blobs
+            .len()
+            .try_into()
+            .expect("live blob count fits into i64");
+
+        let current_batch_blobs: i64 = {
+            let mut current_batch_blobs = HashSet::new();
+            state.collections.trace.map_batches(|b| {
+                for part in b.parts.iter() {
+                    current_batch_blobs.insert(part.key.to_owned());
+                }
+            });
+            current_batch_blobs
+                .len()
+                .try_into()
+                .expect("current blob count fits into i64")
+        };
+
+        machine.metrics.state.current_blobs.set(current_batch_blobs);
```

Comment:
hmm, this is going to set the metric for just this shard, so the value of the metric will hop around based on which shard was the last one to have gc run

I think we need to make it per-shard? nothing else jumps out at me. @pH14 any ideas?

Reply:
Ugh, shouldn't have assumed that the state metrics were per-shard. I guess I should move these over to `ShardsMetrics` then as well?

Another possibility would be to assemble all the per-shard metrics into a largish GC log. (Lots of other interesting metrics in debug logs in this file that could be included.) Loki ought to handle it fairly well. Seems fine if this is mostly for active investigation and not passive dashboarding.

Reply:
I'd like to see our logs (and tracing) infra settle down, handle prod traffic, etc before we start relying on them for things. so I'd prefer to keep as much as possible in prom/grafana for now

Reply:
also pretty sure asf specifically said shard_id would be too high cardinality for it to be a tag in loki, anyway 🤷

Reply:
Too high cardinality to _index_, I think! But we should still be able to query on it, as long as we had other tags to filter the volume down. (But sticking to prometheus sounds fine to me.)

Reply:
fair! as a thought exercise, what indexed tags would we use to filter it down?

Reply:
Naively, I'd probably reach for `k8s_namespace` to get down to the environment level. (Even an index on the "canonical log name" - or however asf referred to that idea - might be enough to start getting useful performance.)

But my mental model here is mostly based on Splunk; I should wait until I spend more time with Loki specifically to start having real opinions!